2_pvc_destination_gen/pvc-destination-gen.yml

```diff
@@ -53,22 +38,49 @@
           loop: "{{ storage_classes.resources }}"
 
         - fail:
+            msg: "One or more storage class mappings are invalid. Exiting..."
           when: "{{ (user_storage_classes | difference(available_storage_classes)) | length > 0 }}"
 
-        - set_fact:
-            pvc_data: "{{ lookup('file', pvc_data_filepath) | from_json }}"
-            namespace_data: "{{ lookup('file', namespace_data_filepath) | from_json }}"
+    - tags:
+      - load_vars
+      - stage_a
+      - stage_b
+      set_fact:
+        pvc_data: "{{ lookup('file', pvc_data_filepath) | from_json }}"
+        namespace_data: "{{ lookup('file', namespace_data_filepath) | from_json }}"
+    
+    - tags:
+      - adjust_pvs
+      - stage_a
+      block:
+      - set_fact:
+          unique_nodes: "{{ pvc_data | json_query('[*].pvcs[*].node_name') | flatten | unique }}"
```

Comment:
This PR has some of the fanciest chains of ansible filters I've seen. Cool stuff.